application/src/main/java/org/gxf/soapbridge/soap/clients/Connection.java

```diff
@@ -0,0 +1,61 @@
+// SPDX-FileCopyrightText: Copyright Contributors to the GXF project
+//
+// SPDX-License-Identifier: Apache-2.0
+package org.gxf.soapbridge.soap.clients;
+
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.TimeUnit;
+import org.gxf.soapbridge.application.services.ConnectionCacheService;
+import org.gxf.soapbridge.application.utils.RandomStringFactory;
+
+public class Connection {
+
+  /**
+   * Default length for the connection id, which is the key for the {@link
+   * ConnectionCacheService#cache}.
+   */
+  private static final int CONNECTION_ID_LENGTH = 32;
+
+  private String soapResponse;
+
+  private final String connectionId;
+
+  private final Semaphore responseReceived;
+
+  public Connection() {
+    responseReceived = new Semaphore(0);
+    connectionId = RandomStringFactory.generateRandomString(CONNECTION_ID_LENGTH);
```

Comment:
Kunnen we hier niet beter een UUID gebruiken? dan hebben we meer zekerheid dat we geen colisions krijgen.